core/io/http_client_tcp.cpp

```diff
@@ -506,13 +505,20 @@ Error HTTPClientTCP::poll() {
 						if (s.length() == 0) {
 							continue;
 						}
-						if (s.begins_with("content-length:")) {
-							body_size = s.substr(s.find_char(':') + 1).strip_edges().to_int();
+						const char contentLengthLabel[] = "content-length:";
+						const char transferEncodingLabel[] = "transfer-encoding:";
+						if (s.begins_with(contentLengthLabel)) {
+							const int index = std::size(contentLengthLabel) - 1;
+							Span<char32_t> sp = Span<char32_t>(s.ptr() + index, s.length() - index);
+							sp = String::strip_edges_span(sp, true, true);
+							body_size = String::to_int(sp.begin(), sp.size());
 							body_left = body_size;
 
-						} else if (s.begins_with("transfer-encoding:")) {
-							String encoding = header.substr(header.find_char(':') + 1).strip_edges();
-							if (encoding == "chunked") {
+						} else if (s.begins_with(transferEncodingLabel)) {
+							const int index = std::size(transferEncodingLabel) - 1;
+							Span<char32_t> sp = Span<char32_t>(s.ptr() + index, s.length() - index);
+							sp = String::strip_edges_span(sp, true, true);
+							if (String("chunked") == sp) {
```

Comment:
This shouldn't need to allocate a `String`. However, my [equality operator PR](https://github.com/godotengine/godot/pull/104280) is still outstanding, so I think it's acceptable for the moment.

Reply:
yeah, it would be better to not have to construct the string, but we can't right now. We can update this if your PR is merged.